lib/jsonapi/view.ex

```diff
@@ -94,23 +96,38 @@ defmodule JSONAPI.View do
         do: serialize(__MODULE__, models, conn)
 
       def url_for(nil, nil) do
-        "#{@namespace}/#{type()}"
+        case @host do
+        ""   -> "#{@namespace}/#{type()}"
+        host -> "#{host}#{@namespace}/#{type()}"
+        end
       end
 
       def url_for(data, nil) when is_list(data) do
-        "#{@namespace}/#{type()}"
+        case @host do
+          ""   -> "#{@namespace}/#{type()}"
+          host -> "#{host}#{@namespace}/#{type()}"
+        end
       end
 
       def url_for(data, nil) do
-        "#{@namespace}/#{type()}/#{id(data)}"
+        case @host do
+          ""   -> "#{@namespace}/#{type()}/#{id(data)}"
+          host -> "#{host}#{@namespace}/#{type()}/#{id(data)}"
+        end
       end
 
       def url_for(data, %Plug.Conn{}=conn) when is_list(data) do
-        "#{Atom.to_string(conn.scheme)}://#{conn.host}#{@namespace}/#{type()}"
+        case @host do
+          ""   -> "#{Atom.to_string(conn.scheme)}://#{conn.host}#{@namespace}/#{type()}"
```

Comment:
If we're going to make the host configurable, shouldn't the scheme be as well?  If you're using SSL termination at your LB your application is going to receive HTTP requests and generate HTTP URLs.